sandy/core/cov.py

```diff
@@ -621,9 +621,55 @@ def from_stdev(cls, std):
         var = std_ * std_
         return CategoryCov.from_var(var)
 
-    def _gls_sensitivity(self, S, Vy, threshold=None):
+    def _gls_general_sensitivity(self, S, Vy, threshold=None):
         """
-        Method to obtain sensitivity according to GLS
+        Method to obtain general sensitivity according to GLS
+
+        Parameters
+        ----------
+        S : 2D iterable
+            Sensitivity square matrix (MXM).
+        Vy : 1D iterable
+            Extra Covariance vector (MX1).
+        threshold : `int`, optional
+            threshold to avoid numerical fluctuations. The default is None.
+
+        Returns
+        -------
+        `GLS`
+            GLS sensitivity for a given Vy and S.
+
+        Example
+        -------
+        >>> S = np.array([[1, 2], [3, 4]])
+        >>> sen = sandy.CategoryCov.from_var([1, 1])
+        >>> sen._gls_general_sensitivity(S, np.array([1, 1]))
+                      0	          1
+        0	-2.00000e-01	2.28571e-01
+        1	2.00000e-01	5.71429e-02
+
+        >>> S = pd.DataFrame([[1, 2], [3, 4]], columns =[1, 2],index=[3, 4])
+        >>> sen = sandy.CategoryCov.from_var([1, 1])
+        >>> sen._gls_general_sensitivity(S,pd.Series([1, 1], index=[1, 2]))
+                      1	          2
+        1	-2.00000e-01	2.28571e-01
+        2	2.00000e-01	5.71429e-02
+        """
+        columns = pd.DataFrame(S).columns
+        S_ = pd.DataFrame(S).values
+        Vy_ = pd.DataFrame(np.diag(pd.Series(Vy))).values
+        # GLS_sensitivity:
+        Vx = self.data.values
+        M = S_.T.dot(Vx).dot(S_) + Vy_
+        M_inv = sandy.CategoryCov(M).invert()
+        sensitivity = Vx.dot(S_).dot(M_inv.data.values)
+        if threshold is not None:
+            sensitivity[sensitivity < threshold] = 0
+        return pd.DataFrame(sensitivity, index=columns, columns=columns)

```

Comment:
no! index and columns should be the same as in S, and they might differ one from the other!